java/gateway-core/src/main/java/com/avaulta/gateway/rules/JsonSchemaFilterUtils.java

```diff
@@ -65,9 +65,79 @@ private Object filterBySchema(String path, JsonNode provisionalOutput, JsonSchem
+        } else if (schema.hashOneOf()) {
+            // Get first schema with matches its inner condition.
+            // See https://json-schema.org/understanding-json-schema/reference/combining.html#oneof
+            // NOTE: If is expected that the "oneOf" candidate should hava an if-else-then or if-then nodes
+            // inside, otherwise the condition will not be evaluated and only the first occurrence appearing in the list
+            // will be chosen
+            for (JsonSchemaFilter oneOfCandidate : schema.getOneOf()) {
+                Object result = filterBySchema(path, provisionalOutput, oneOfCandidate, root, redactionsMade);
+
+                if (!(result instanceof NotMatchedConstant)) {
+                    return result;
+                }
+            }
+
+            return null;
+        } else if (schema instanceof ConditionJsonSchema) {
+            // Conditions are schemas without no type definition
+            // Only one property are supported by conditions. See https://json-schema.org/understanding-json-schema/reference/conditionals.html#if-then-else for futher details;
+            // in case of more than one property needs to be used to match a condition they should be included inside on allOf/anyOf properties
+            Map.Entry<String, JsonSchemaFilter> property = schema.getProperties().entrySet().stream().findFirst()
+                    .orElseThrow(() -> new IllegalArgumentException("Invalid schema, a single property is expected"));
+
+            String key = property.getKey();
+
+            return filterBySchema(path + "." + key, provisionalOutput.get(property.getKey()), schema.getProperties().get(property.getKey()), root, redactionsMade);
+        } else if (schema instanceof ThenJsonSchema) {
+            // NOTE: Using a LinkedHashMap to keep the fields in the same order
+            // they appear defined; otherwise even the final order it seems to be deterministic
+            // is might not be the same
+            Map<String, Object> filtered = new LinkedHashMap<>();
+            provisionalOutput.fields().forEachRemaining(entry -> {
+                String key = entry.getKey();
+                JsonNode value = entry.getValue();
+                JsonSchemaFilter propertySchema = schema.getProperties().get(key);
+
+                if (propertySchema == null) {
+                    log.info("Redacted " + path + "." + key + " because it was not in schema");
+                    redactionsMade.add(path + "." + key);
+                } else {
+                    Object filteredValue = filterBySchema(path + "." + key, value, propertySchema, root, redactionsMade);
+                    filtered.put(key, filteredValue);
+                }
+            });
+
+            return filtered;
         } else if (schema.hasType()) {
-            //must have explicit type
 
+            if (schema.hasIf()) {
+                Object conditionResult = filterBySchema(path, provisionalOutput,
```

Comment:
Adding conditional rules following the [JSON standard spec](https://json-schema.org/understanding-json-schema/reference/conditionals.html#id6). 
If a schema contains an if, it will check the expression. At the moment, `constant` has been added to included in the check as it is the only currently required. Condition can return anything or an instance of `NotMatchedConstant` which in this case, the parent should handle about what to do